genome_designer/main/xhr_handlers.py

```diff
@@ -1041,6 +1045,97 @@ def get_contigs(request):
 
 @login_required
 @require_GET
+def contigs_has_insertion_location(request):
+    """Get list of Contigs for the provided Project uid.
+    """
+    # Parse the GET params.
+    request_data = json.loads(request.GET['data'])
+    contig_uid_list = request_data.get('contigUidList', None)
+
+    # Search metadata of filtered contigs for non-null
+    # insertion location keys
+    has_insertion_location = False
+    for c in Contig.objects.filter(uid__in=contig_uid_list):
+        if c.metadata.get('insertion_sequence_endpoints', None):
+            has_insertion_location = True
+            break
+
+    result = {'has_insertion_location': has_insertion_location}
+    return HttpResponse(json.dumps(result), content_type='application/json')
+
+
+@login_required
+@require_POST
+def contigs_find_insertion_location(request):
+    request_data = json.loads(request.body)
+    contig_uid_list = request_data.get('contigUidList', [])
+
+    result = {}
+    for contig in Contig.objects.filter(uid__in=contig_uid_list):
+        ref_genome = contig.parent_reference_genome
+        contig_fasta = get_dataset_with_type(contig,
+                Dataset.TYPE.REFERENCE_GENOME_FASTA).get_absolute_location()
+        with open(contig_fasta) as fh:
+            seqrecord = SeqIO.parse(fh, 'fasta').next()
+            insertion_data = find_contig_insertion_site(
```

Comment:
We'll probably want to make this asynchronous vs celery, depending on how slow it is with full-sized data. Need to QA.